controllers/job.go

```diff
@@ -181,13 +185,34 @@ func (controller *JobController) Post(w http.ResponseWriter, r *http.Request, pa
 		return
 	}
 
-	updateData := struct{ NextState string }{}
+	updateData := struct {
+		NextState string
+		Timeout   time.Duration
+	}{}
 	err := json.NewDecoder(r.Body).Decode(&updateData)
 	if err != nil {
 		writeErr(w, err)
 		return
 	}
 
+	new_timeout := false
+	if updateData.Timeout != 0 {
+		new_timeout = true
+		job.Timeout = updateData.Timeout
+	}
+
+	if updatingTimeoutForNonInflightState := new_timeout &&
```

Comment:
Why is this assignment required?

Reply:
Bhaia, it is not necessary, but I thought it might help in readability about what the logical operations are trying to achieve. 
Do you think I should remove it?

Reply:
I think we should remove it if its not reused